lib/init.coffee

```diff
@@ -78,127 +81,76 @@ module.exports =
       helpers = require 'atom-linter'
       voucher = require 'voucher'
       fs = require 'fs'
-      if @verboseLogging
-        @_log 'requiring modules finished.'
+      @_log 'requiring modules finished.'
 
-    if @verboseLogging
-      @_log 'providing linter, examining javac-callability.'
+    @_log 'providing linter, examining javac-callability.'
 
     grammarScopes: ['source.java']
     scope: 'project'
     lintOnFly: false       # Only lint on save
+
     lint: (textEditor) =>
       filePath = textEditor.getPath()
       wd = path.dirname filePath
       searchDir = @getProjectRootDir() || path.dirname filePath
+
       # Classpath
       cp = ''
 
-      if @verboseLogging
-        @_log 'starting to lint.'
+      # Sourcepath - this is a fair default for many Java projects
+      sp = 'src/main/java/'
+
+      @_log 'starting to lint.'
 
       # Find project config file if it exists.
-      cpConfig = @findClasspathConfig(wd)
+      cpConfig = @findConfigFile(wd, @classpathFilename)
       if cpConfig?
         # Use the location of the config file as the working directory
-        wd = cpConfig.cfgDir
+        wd = cpConfig.dir
         # Use configured classpath
-        cp = cpConfig.cfgCp
+        cp = cpConfig.content
         # Use config file location to import correct files
         searchDir = wd
 
+      spConfig = @findConfigFile(wd, @sourcepathFilename)
+      if spConfig?
+        sp = spConfig.content
+
       # Add extra classpath if provided
       cp += path.delimiter + @classpath if @classpath
 
       # Add environment variable if it exists
       cp += path.delimiter + process.env.CLASSPATH if process.env.CLASSPATH
 
-      if @verboseLogging
-        @_log 'start searching java-files with "',
-          searchDir,
-          '" as search-directory.'
+      @_log 'start searching java-files with "', searchDir, '" as search-directory.'
 
       lstats = fs.lstatSync searchDir
 
-      atom.project.repositoryForDirectory(
-        new Directory(searchDir, lstats.isSymbolicLink())
-      )
-        .then (repo) =>
-          @getFilesEndingWith searchDir,
```

Comment:
Whatever you call the next version is up to you. For most users I think the number is irrelevant though. The notification to update will be there, wether it's to `1.9.5` or `2.0.0`. If people think this is a bad change we'd have to deal with it either way.

Semantical versioning suggests it should be named `2.0.0` of course. But dealing with users will have to be done either way.